packages/reference/src/common/SortableLinkList.tsx

```diff
@@ -26,55 +30,118 @@ type SortableContainerChildProps<IType> = Pick<
 > & {
   item: IType;
   index: number;
-  DragHandle?: typeof DragHandle;
+  DragHandle?: (props: { drag: React.ReactElement }) => React.ReactElement;
 };
 
 type SortableLinkListProps<T> = ReferenceEditorProps & {
   items: T[];
   isDisabled: boolean;
   children: (props: SortableContainerChildProps<T>) => React.ReactElement;
-
+  sortingStrategy?: SortingStrategy;
+  onSortStart?: (event: DragStartEvent) => void;
+  onSortEnd?: ({ oldIndex, newIndex }: { oldIndex: number; newIndex: number }) => void;
   className?: string;
 };
 
-const DragHandle = (props: { drag: React.ReactElement }) => {
-  const SortableDragHandle = SortableHandle(() => props.drag);
-  return <SortableDragHandle />;
-};
+interface SortableLinkProps<T extends { sys: any }> {
+  id: string;
+  items: T[];
+  item: T;
+  isDisabled?: boolean;
+  index: number;
+  children: (props: SortableContainerChildProps<T>) => React.ReactElement;
+}
 
-const SortableLink = SortableElement((props: { children: React.ReactElement }) => (
-  <div className={styles.item}>{props.children}</div>
-));
+const SortableLink = <T extends { sys: any }>({
+  id,
+  items,
+  item,
+  isDisabled = false,
+  index,
+  children,
+}: SortableLinkProps<T>) => {
+  const { listeners, setNodeRef, setActivatorNodeRef, transform, transition, isDragging } =
+    useSortable({ id });
+  const style = {
+    transform: transform ? CSS.Transform.toString({ ...transform, scaleX: 1, scaleY: 1 }) : '',
+    transition,
+  };
+
+  const DragHandle = (props: { drag: React.ReactElement }) => {
```

Comment:
Can't we use `https://docs.dndkit.com/api-documentation/draggable` and extract it into it's own component?
Currently it will be recreated through each re-rendering

Reply:
Also on the useSortable documentation:
```
function SortableItem(props) {
  const {arguments, listeners, setNodeRef} = useDraggable({
    id: props.id,
  });
  
  return (
    <li ref={setNodeRef}>
      {/* ... */}
      <button {...listeners} {...arguments}>Drag handle</button>
    </li>
  );
}
```

Reply:
No, because we use `useSortable` both in handle (see [`setActivatorNodeRef`](https://docs.dndkit.com/presets/sortable/usesortable#activator)) and in card itself.

Or maybe I'm missing something here?

Reply:
Yeah, I get it, but it will be tricky to pass props there as DragHandle (that comes from EntryCard component) accepts only function with props: { drag }. How about just using `useCallback`?

Reply:
Yeah or `useMemo`, also fine if one of this works then 👍